app/models/merchant.rb

```diff
@@ -6,4 +6,7 @@ def self.random
     order("RANDOM()").first
   end
 
+  def favorite_customer
+    Customer.joins(:transactions).where(transactions: {result: "success"}, 	invoices: {merchant_id: self.id}).group('customers.id').order("count(customers.id) 	desc").limit(1).first
+  end
```

Comment:
Don't forget to update this to use the scope `Transaction.success` that is on the other branch!